File: VB/EditReferenceProperiesInBatchEditMode.Module/BusinessObjects/LargeDataDemo.vb
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Xpo

<DefaultClassOptions>
Public Class LargeDataDemo
    Inherits BaseObject

    Public Sub New(ByVal session As Session)
        MyBase.New(session)
    End Sub

    Private _name As String

    Public Property Name As String
        Get
            Return _name
        End Get
        Set(ByVal value As String)
            SetPropertyValue(Of String)(NameOf(Name), _name, value)
        End Set
    End Property

    Private _largeData As LargeData

    Public Property LargeData As LargeData
        Get
            Return _largeData
        End Get
        Set(ByVal value As LargeData)
            SetPropertyValue(NameOf(LargeData), _largeData, value)
        End Set
    End Property
End Class

Public Class LargeData
    Inherits BaseObject

    Public Sub New(ByVal session As Session)
        MyBase.New(session)
    End Sub

    Private _name As String

    Public Property Name As String
        Get
            Return _name
        End Get
        Set(ByVal value As String)
            SetPropertyValue(Of String)(NameOf(Name), _name, value)
        End Set
    End Property
End Class

File: CS/EditReferenceProperiesInBatchEditMode.Module/BusinessObjects/LargeDataDemo.cs
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Xpo;

namespace EditReferenceProperiesInBatchEditMode.Module.BusinessObjects {
    [DefaultClassOptions]
    public class LargeDataDemo : BaseObject {
        public LargeDataDemo(Session session) : base(session) { }
        private string name;
        public string Name {
            get {
                return name;
            }

            set {
                SetPropertyValue<string>(nameof(Name), ref name, value);
            }
        }
        private LargeData largeData;
        public LargeData LargeData {
            get {
                return largeData;
            }

            set {
                SetPropertyValue(nameof(LargeData), ref largeData, value);
            }
        }
    }
    public class LargeData : BaseObject {
        public LargeData(Session session) : base(session) { }
        private string name;
        public string Name {
            get {
                return name;
            }

            set {
                SetPropertyValue<string>(nameof(Name), ref name, value);
            }
        }
    }
}
